src/index/pretransform.rs

```diff
@@ -89,7 +89,7 @@ impl<I> NativeIndex for PreTransformIndexImpl<I> {
     }
 }
 
-impl<I> FromInnerPtr for PreTransformIndexImpl<I> {
+impl<IndexImpl> FromInnerPtr for PreTransformIndexImpl<IndexImpl> {
```

Comment:
It's existing type `IndexImpl`.

Reply:
Hmm. In that case it should be this, no? (note the lack of type parameters)

```rust
impl FromInnerPtr for PreTransformIndexImpl<IndexImpl> {
```

Reply:
It does not work. It needs type parameter

Reply:
We tried it https://github.com/Enet4/faiss-rs/pull/60/commits/a07ca9b7b36897381fe14c5d556018e1ec04c46d

Reply:
OK, so that error was:

```none
error[E0277]: the trait bound `RefineFlatIndexImpl<BI>: index::FromInnerPtr` is not satisfied
   --> src/index/refine_flat.rs:225:20
    |
225 | impl<BI: TryClone> TryClone for RefineFlatIndexImpl<BI> {}
    |                    ^^^^^^^^ the trait `index::FromInnerPtr` is not implemented for `RefineFlatIndexImpl<BI>`
    |
note: required by a bound in `index::TryClone`
   --> src/index/mod.rs:339:21
    |
339 | pub trait TryClone: FromInnerPtr {
    |                     ^^^^^^^^^^^^ required by this bound in `index::TryClone`
help: consider introducing a `where` clause, but there might be an alternative better way to express this requirement
    |
[22](https://github.com/Enet4/faiss-rs/actions/runs/3280362393/jobs/5401039258#step:6:23)5 | impl<BI: TryClone> TryClone for RefineFlatIndexImpl<BI> where RefineFlatIndexImpl<BI>: index::FromInnerPtr {}
    |                                                         ++++++++++++++++++++++++++++++++++++++++++++++++++
```

The error message already suggests a way to fix the problem, but the way I see it, the `TryClone` trait is overly restrictive right now due to its default implementation of `try_clone`. So we can either do what the error suggests or move the default implementation onto those that actually depend on it. The steps would be:

1. Create a helper function `try_clone_from_inner_ptr` with the default implementation in `TryClone::try_clone`.
2. Remove default implementation body of method `TryClone::try_clone`.
3. In each `TryClone` implementation with a missing `try_clone`, add the implementation to make them call `try_clone_from_inner_ptr`.

Adding a type parameter might make the error go away, but it's not what we want.